modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/runtime/CaseInstanceHelperImpl.java

```diff
@@ -421,6 +472,178 @@ protected CaseInstanceEntity createCaseInstanceEntityFromDefinition(CommandConte
+        planItemInstanceEntity.setCompletedTime(historicPlanItemInstance.getCompletedTime());
+        planItemInstanceEntity.setOccurredTime(historicPlanItemInstance.getOccurredTime());
+        planItemInstanceEntity.setTerminatedTime(historicPlanItemInstance.getTerminatedTime());
+        planItemInstanceEntity.setExitTime(historicPlanItemInstance.getExitTime());
+        planItemInstanceEntity.setEndedTime(historicPlanItemInstance.getEndedTime());
+        planItemInstanceEntity.setStartUserId(historicPlanItemInstance.getStartUserId());
+        planItemInstanceEntity.setReferenceId(historicPlanItemInstance.getReferenceId());
+        planItemInstanceEntity.setReferenceType(historicPlanItemInstance.getReferenceType());
+        planItemInstanceEntity.setEntryCriterionId(historicPlanItemInstance.getEntryCriterionId());
+        planItemInstanceEntity.setExitCriterionId(historicPlanItemInstance.getExitCriterionId());
+        planItemInstanceEntity.setFormKey(historicPlanItemInstance.getFormKey());
+        planItemInstanceEntity.setExtraValue(historicPlanItemInstance.getExtraValue());
+        planItemInstanceEntity.setTenantId(historicPlanItemInstance.getTenantId());
+
+        planItemInstanceEntityManager.insert(planItemInstanceEntity);
+        return planItemInstanceEntity;
+    }
+
+    /**
+     * Creates new variables in the runtime according the history of the provided case instance.
+     *
+     * @param historicCaseInstance the historic case instance to copy its variables back to the runtime
+     * @param newCaseInstance the new case instance where the variables will be attached to
+     * @return the map of the created variables
+     */
+    protected Map<String, VariableInstanceEntity> createCaseVariablesFromHistoricCaseInstance(HistoricCaseInstance historicCaseInstance, CaseInstanceEntity newCaseInstance) {
+        VariableService variableService = cmmnEngineConfiguration.getVariableServiceConfiguration().getVariableService();
+        VariableTypes variableTypes = cmmnEngineConfiguration.getVariableTypes();
+        List<HistoricVariableInstance> variables = cmmnEngineConfiguration.getCmmnHistoryService()
+            .createHistoricVariableInstanceQuery()
+            .caseInstanceId(historicCaseInstance.getId())
+            .list();
+
+        if (variables != null) {
+            Map<String, VariableInstanceEntity> newVars = new HashMap<>(variables.size());
+            for (HistoricVariableInstance variable : variables) {
+                VariableInstanceEntity newVariable = variableService.createVariableInstance(
+                    variable.getVariableName(), variableTypes.getVariableType(variable.getVariableTypeName()), variable.getValue());
+
+                newVariable.setId(variable.getId());
+                newVariable.setScopeId(newCaseInstance.getId());
+                newVariable.setScopeType(variable.getScopeType());
+                newVariable.setSubScopeId(variable.getSubScopeId());
```

Comment:
don't copy, if there is a sub-scope set as it would mean to copy local vars like from stages, etc which we don't need